src/mlpack/bindings/cli/parse_command_line.hpp

```diff
@@ -34,122 +35,72 @@ PARAM_FLAG("version", "Display the version of mlpack.", "V");
-  std::map<std::string, std::string> boostNameMap;
-  for (IteratorType it = parameters.begin(); it != parameters.end(); ++it)
+  std::map<std::string, util::ParamData>& parameters = IO::Parameters();
+  using ItType = std::map<std::string, util::ParamData>::iterator;
+
+  for (ItType it = parameters.begin(); it != parameters.end(); ++it)
   {
     // Add the parameter to desc.
-    const util::ParamData& d = it->second;
-    CLI::GetSingleton().functionMap[d.tname]["AddToPO"](d, NULL,
-        (void*) &desc);
-
-    // Generate the name the user passes on the command line.
-    std::string boostName;
-    CLI::GetSingleton().functionMap[d.tname]["MapParameterName"](d, NULL,
-        (void*) &boostName);
-    boostNameMap[boostName] = d.name;
+    util::ParamData& d = it->second;
+    IO::GetSingleton().functionMap[d.tname]["AddToCLI11"]
+      (d, NULL, (void*) &app);
   }
 
   // Mark that we did parsing.
-  CLI::GetSingleton().didParse = true;
+  IO::GetSingleton().didParse = true;
 
   // Parse the command line, then place the values in the right place.
   try
   {
-    basic_parsed_options<char> bpo(parse_command_line(argc, argv, desc));
-
-    // Iterate over all the options, looking for duplicate parameters.  If we
-    // find any, remove the duplicates.  Note that vector options can have
-    // duplicates, so we check for those with max_tokens().
-    for (size_t i = 0; i < bpo.options.size(); ++i)
+    try
     {
-      for (size_t j = i + 1; j < bpo.options.size(); ++j)
-      {
-        if ((bpo.options[i].string_key == bpo.options[j].string_key) &&
-            (desc.find(bpo.options[i].string_key,
-                       false).semantic()->max_tokens() <= 1))
-        {
-          // If a duplicate is found, check to see if either one has a value.
-          if (bpo.options[i].value.size() == 0 &&
-              bpo.options[j].value.size() == 0)
-          {
-            // If neither has a value, we'll consider it a duplicate flag and
-            // remove the duplicate.  It's important to not break out of this
-            // loop because there might be another duplicate later on in the
-            // vector.
-            bpo.options.erase(bpo.options.begin() + j);
-            --j; // Fix the index.
-          }
-          else
-          {
-            // If one or both has a value, produce an error and politely
-            // terminate.  We pull the name from the original_tokens, rather
-            // than from the string_key, because the string_key is the parameter
-            // after aliases have been expanded.
-            Log::Fatal << "\"" << bpo.options[j].original_tokens[0] << "\" is "
-                << "defined multiple times." << std::endl;
-          }
-        }
-      }
+      app.parse(argc, argv);
+    }
+    catch (const CLI::ArgumentMismatch& err)
+    {
+      Log::Fatal << "An option is defined multiple times." 
+                 << app.exit(err) << std::endl; 
+    }
```

Comment:
Also we should catch `OptionNotFound`:

```
$ bin/mlpack_pca
terminate called after throwing an instance of 'CLI::OptionNotFound'
  what():  input_file not found
Aborted
```

I think this could work:

```suggestion
    catch (const CLI::OptionNotFound& onf)
    {
      Log::Fatal << "Required option --" << app.exit(onf) << "!" << std::endl;
    }
```
(I think that will line up the `--` with the option name correctly...)